### ISSUE TITLE
Abandoned project?

### ISSUE THREAD

**Issue by desfire** [serghey-rodin/vesta#1330]
Hi,
Is Vesta abandoned? 1 year without new releases, that's insane, many bugs, is there at least a fork which more updates?

**Comment by robindirksen1** [comment]
Who can update the public repo, only @serghey-rodin ? @dpeca

**Comment by madeITBelgium** [comment]
I, @dpeca, @naumov-socolov and some others have access to the GitHub repo. But we do not have access to the vesta servers to build and publish a new release.

I can build a release for centos (I do it already on my own fork).

**Comment by robindirksen1** [comment]
So whe have to wait on serghey to gets a new update? 

@madeITBelgium do you know when it will be?

**Comment by dpeca** [comment]
Many bugs are fixed on github repo... but only Serghey has access to c.vestacp.com so without him we are unable to update official repo on official server with bugfixes from github.

**Comment by robindirksen1** [comment]
So when will be the next release?

**Comment by desfire** [comment]
So nobody can reach him to update the c.vestacp.com, vesta is the best free control panel and a year without updates is actually really bad, there are a lot of errors reported here and in the forum

**Comment by anton-reutov** [comment]
Sergey said to me that he is already preparing a new release.

**Comment by lukapaunovic** [comment]
You can easily update vesta directly from github like @dpeca said.
(Install GIT)

`cd /root
git clone https://github.com/serghey-rodin/vesta.git
yes | cp -rf /root/vesta/* /usr/local/vesta`

**Comment by anton-reutov** [comment]
@lukapaunovic, we are no recommend to do this. Only in exceptional cases.

**Comment by lukapaunovic** [comment]
@anton-reutov why not? If I do not do this I have to fix a lot of issues manually.

**Comment by madeITBelgium** [comment]
Because the GitHub version is not tested completely. There might be unknown bugs. That can cause other problems. It's a development branch.

**Comment by dpeca** [comment]
That's why I forked this repo, and update that repo by accepting commits from this master repo, which means I review them and almost know what is changed and that way I know if some commit is suspicious...
Then I update my servers from that my repo...

**Comment by lukapaunovic** [comment]
So it's safe to `git clone https://github.com/dpeca/vesta.git` then? :)

**Comment by n1trux** [comment]
I tried to fork this into https://github.com/n1trux/hestia but right as I tried to remove dependencies on c.vestacp.com, they pushed a new spate of commits to the "official" repo.

I'd really appreciate people helping me over there. @dpeca is it possible that we work together on this? Maybe you can include all of your changes into my repo or vice versa?

**Comment by dpeca** [comment]
@lukapaunovic if you believe that I will not update 20 dedicated/vps production servers with untested code, then yes.
But again, I dont want to be responsible for other servers, so everyone should update on his own responsibility.

@n1trux I don't have ''my changes'' that is separately added to my fork.
All changes that I made are pushed to this master official repo, and I dont have a wish to work on separated project, it will be wasting a time.
My fork is just a "gate" that help me review commits from master before they end on my production servers.
On master repo I can not "approve" nor deny commits, so with my fork I just put commit "on hold" before they enter to my repo.

We should stay gathered around ONE project.
Thanks to git forks you can have your own "little modified" source of VestaCP, but I advice you to keep compatibility with master repo, you can update your fork with new commits (from master) just like I do with my "gate" fork.

And, by the way, you can make your own "c.vestacp.com" by tracing "wget" in installer script, you can find a list of files in installer script... then simply download it to your server.

**Comment by dpeca** [comment]
And probably 99% of files on **c.vestacp.com** are from **installer/** folder from github repo...

**Comment by lukapaunovic** [comment]
Just to clarify, I'm not insane to `git `random code from random people repos, but I trust @dpeca cuz' i know him.

**Comment by dalu** [comment]
vesta is a mess
good intentions but lacking experience and knowledge
it's very amateur
it has no good concept
packages? what if one customer wants 2 packges, do you create 2 customers?
why limit yourself with packages. packages if at all should serve as templates for actual values.
that's just one example.
it's really not very well thought out.
the installer is nice. yeah. you can get it up and running quickly.
mail works.
but ...
inexperience shows.
I wish it was different.
I needed a quick solution and that's why I used it.

**Comment by lukapaunovic** [comment]
@dalu if you don't like vesta you can use <a href="http://www.webmin.com/virtualmin.html">Virtualmin and Webmin</a>
Also, there is <a href="https://github.com/mustafaramadhan/kloxo/tree/dev">Kloxo-MR</a> which is a continued project by one of the original Kloxo developers.

**Comment by edirpedro** [comment]
Does he needs a crowdfunding to put another developers to work here?

**Comment by dpeca** [comment]
I think that crowdfunding is not necessary, since here are many developers that helps the project with push requests and other activity on github... they already have a interest for developing VestaCP because they use VestaCP (for free) on their own servers and makes money from those servers.

I think that main "stumbling stone" is a fact that only @serghey-rodin can push new code to **c.vestacp.com** repo.
Since he was very busy this year, nobody could update official repo with bugfixes.

Last year we had a deal that we will push mini-updates every month, but that deal failed because nobody can push new code to official repo.

I believe @serghey-rodin should choose one more person that he can trust and to give him SSH access to **c.vestacp.com**.
We probably can not release 0.9.8-**18** without Serghey, but definitely we can release 0.9.8-17-**1**, 0.9.8-17-**2**, 0.9.8-17-**3** and so on... bugfix releases...

I think @serghey-rodin personally knows @naumov-socolov - maybe he knows @anton-reutov too... anybody that Serghey can trust can help us to update official repo with mini-bugfix-releases...

**Comment by madeITBelgium** [comment]
Why not automated? Simple bash script that builds and deploy it. Maybe we can use Jenkins or Travis or some other CI tool to test and auto deploy.

**Comment by dpeca** [comment]
Because sometimes dangerous commits goes to master, we already had that situation in January this year... master branch was broken two months...

**Comment by madeITBelgium** [comment]
I do not mean that we need to do it completely automated. But create a nightly build server. Everyone can test on that version. Create beta release every x weeks. If that is OK someone triggers the official release. 

I already created my own Nightly build process: https://github.com/madeITBelgium/vesta/tree/vestacp-master

We can start with fix the unit tests.
 https://github.com/serghey-rodin/vesta/tree/master/test

**Comment by devs4p** [comment]
I forked this repo just an hour ago and am currently working on it. Just give me list of TODO's and we will work on it together. I will soon invite some of you to join me there.

In the near future, Im planning to open a **mirror repo** of vesta so it'd be easier to publish new versions as I have write-access.

**Comment by lukapaunovic** [comment]
@devs4p @madeITBelgium 
Wanna change something? Then do a pull request. And developers will accept or decline it. 
I do not support having 1000 mirror repositories which were created by random people and every single one of them thinks he knows the best.

**Comment by huloza** [comment]
good ideas, but lets do it now! i can help too.

**Comment by hwcltjn** [comment]
What about creating a simple script that updates 0.9.8-17 with the commits and changes that have been identified as stable by the community?
I think it would be welcomed by those that are not familiar with git.

Had I known that @dpeca created a stable mirror I would have used that instead :)

**Comment by lukapaunovic** [comment]
@hwcltjn The master GitHub repo is continually developed and checked by all contributors and collaborators. It's not just one person, it's a few persons working + dozens _hidden_ persons like me watching the few persons work, like watchdogs.  ;)

**Comment by hwcltjn** [comment]
@lukapaunovic Oh absolutely - I get that. 
But as it has been mentioned a few times in the forums and even in this thread replacing the contents of /usr/local/vesta with the github repo is not advised.

How about creating something that is both safe and easy to use?
Like @dpeca mentioned 0.9.8-17-1 for example.

**Comment by dpeca** [comment]
It's watched by many people, but in january after very dangerous commit it passed 4 days before I noticed that something wrong happened... and I noticed it only because my client accidently triggered bug... wheach means it could pass much more days before we realize that something is wrong.

Current state of master is stable.
But it's now... I can't guarantee for tomorrow, for example...
Only can gurantee that I'll not merge it to my fork without checking what is changed.

**Comment by anton-reutov** [comment]
The first, we are checking the commits on the test servers and only after that accept it
The changes in the commits shouldn't break the existing functionality. We are very strict about this.

**Comment by fengkaijia** [comment]
First thing first, we need a "How to Build".

**Comment by lukapaunovic** [comment]
@fengkaijia @serghey-rodin can maybe give us some info when is he available to push changes to the server.

**Comment by phez** [comment]
Sorry guys, I am a noob. I want to use @dpeca GIT repo on Centos 6. I did this:
1. yum install git
2. git clone https://github.com/dpeca/vesta.git
3. cp -rf /root/vesta/* /usr/local/vesta
4. cd /usr/local/vesta/install
5. bash vst-install.sh

Is it correct?

**Comment by anton-reutov** [comment]
@phez, its not correct, because official installer vst-install.sh will overwrite third-party files

**Comment by phez** [comment]
Could you please tell me correct method?

**Comment by fengkaijia** [comment]
@phez I believe u should first run vst-install.sh then do the git clone and cp. But need others to confirm.

See, this is the tricky part. The current setup literally prevents the project from been fork, because you still rely on the official installer to install the original version.

**Comment by dpeca** [comment]
Only correct 5. step.
Insted of `bash vst-install.sh` run `bash vst-install-rhel.sh`

After installing you must once again run:
```
cd /root
rm -rf /root/vesta
git clone https://github.com/dpeca/vesta.git
yes | cp -rf /root/vesta/* /usr/local/vesta
```

Second thing, I dont recommend CentOS6.
Centos7 is fine.

Third thing - I must repeat once again - it's deffinetelly less bugs on github version - - but - I dont want to be responsible for other servers, so everyone should use code from github on his own responsibility.

**Comment by anton-reutov** [comment]
Yesterday, i sent to Serghey Rodin the ten most important bugs from the bug tracker. I think he will fix them on Monday or Tuesday

**Comment by phez** [comment]
Thanks @dpeca 
Can you confirm why you said this:
"Second thing, I dont recommend CentOS6.
Centos7 is fine"

Is this just for security purposes?

**Comment by dpeca** [comment]
No.
Vesta just works better with CentOS7, because all CentOS developers mostly use CentOS7, probably very small amount of developers checks how stuff works on CentOS6...

**Comment by huloza** [comment]
CentOS 6 has very outdated software.

**Comment by phez** [comment]
Thank you Team. I am not installing Centos7. After install I will do the following (please correct if wrong).

yum install git
git clone https://github.com/dpeca/vesta.git
cp -rf /root/vesta/* /usr/local/vesta
cd /usr/local/vesta/install
bash vst-install-rhel.sh

Shutdown -r now

cd /root
rm -rf /root/vesta
git clone https://github.com/dpeca/vesta.git
yes | cp -rf /root/vesta/* /usr/local/vesta

**Comment by jodumont** [comment]
@dpeca do you do the same checks on the Ubuntu version ?

**Comment by anton-reutov** [comment]
Warning. All the actions you doing at your own risk. We cant responsible for the correct operation of the software after the changes. Official support is not provided.

**Comment by phez** [comment]
@anton-reutov I/We understand. If you can guide for the install from GIT it would be appreciated.

**Comment by jodumont** [comment]
`doing at your own risk`

@anton-reutov that's the beauty and the darkside of the open source :)

**Comment by dpeca** [comment]
@jodumont 
I'm afraid that Ubuntu support is not so good as we would like.
I personally checks my own commits on all three distributions (Debian, CentOS, Ubuntu), but it looks like that old code wasn't checked always on Ubuntu, or, at least, Vesta developers mostly fix bugs on CentOS and Debian, less on Ubuntu.
When I entered the developers team, I said that I will mostly cover Debian bugs, because I use Debian on my servers.
I will be very happy if we have developer that is covering Ubuntu like I'm covering Debian.

**Comment by anton-reutov** [comment]
@phez 
```
curl -O http://vestacp.com/pub/vst-install.sh
bash vst-install.sh

cd $(mktemp -d)
git clone git://github.com/serghey-rodin/vesta.git
/bin/cp -rf vesta/* /usr/local/vesta/
```

**Comment by dpeca** [comment]
But that way old installer will be used, @anton-reutov 
If somebody already decided to use git version of code, then better to use installer from git.

If someone want to use installer from git (that is newer version), then:
```
cd $(mktemp -d)
git clone git://github.com/serghey-rodin/vesta.git
cd vesta/install/vst-install-rhel.sh #or other distro
cd ../../
yes | cp -rf vesta/* /usr/local/vesta
```

**Comment by phez** [comment]
Thank you @anton-reutov and @dpeca 
Very much appreciated !!

**Comment by darkworks** [comment]
CentOS is another name of windows use a lot of resources , Debian is best , use less resources

**Comment by desfire** [comment]
Has also anyone tried to add Hiawatha in Vesta? Is a more secure and faster than nginx

**Comment by darkworks** [comment]
also how about integrating  duplicati in vesta 

> https://github.com/duplicati/duplicati

**Comment by dpeca** [comment]
In next period (until we release v18) I don't recommend updating from master branch, because a lot of code is changing, and there is possibility of bugs.
I will not update my fork until we test everything.

**Comment by lukapaunovic** [comment]
Who da man?! **'You da man!**

**Comment by anton-reutov** [comment]
@lukapaunovic what is mean? ))

**Comment by lukapaunovic** [comment]
@anton-reutov  **'You da man! too**

**Comment by anton-reutov** [comment]
Who wants to test how the domains is working on the code from github ?

**Comment by dpeca** [comment]
I planned to do that as soon as I can, but can not promise when it will be... in next 7 days I hope.

**Comment by hwcltjn** [comment]
Domains? What’s changed?

**Comment by anton-reutov** [comment]
@dpeca Need as soon as possible. The new version should be release soon
@hwcltjn templates

**Comment by azzronulation** [comment]
Any official word on a new release?

**Comment by anton-reutov** [comment]
no one wants to help :))
ок

**Comment by azzronulation** [comment]
@anton-reutov I’d offer, but our server is for a live site, can’t risk any downtime. Sorry!

**Comment by gethprime** [comment]
@anton-reutov I'll pull up a test server on a cloud server in a bit. Is there any specific tests that you would like me to perform or should I just check for general issues with domains?

**Comment by anton-reutov** [comment]
@amaisonneuve do you have account on our forum ?
https://forum.vestacp.com/

**Comment by hwcltjn** [comment]
@anton-reutov I'm also happy to test the templates on a new server - what are you looking for? My name on the forums is the same as on here.

**Comment by anton-reutov** [comment]
@amaisonneuve the server is ready. Send your login on vestacp forum

**Comment by anton-reutov** [comment]
@hwcltjn the server is ready. I send you PM on vestacp forum

**Comment by dpeca** [comment]
Due to my private obligations, I was unable to do that today... and tomorrow I'll have day of hell :(
I hope I'll have free time tomorrow at night.
Sorry guys :/

**Comment by amaisonneuve** [comment]
Yes amaisonneuve

**Comment by anton-reutov** [comment]
@gethprime do you have the test server for tests ?
@amaisonneuve the server is ready. I sent you PM on vestacp forum

**Comment by anton-reutov** [comment]
@dpeca ok

**Comment by amaisonneuve** [comment]
I'm in

**Comment by desfire** [comment]
I assume Softaculous won't be in this new version right?

**Comment by anton-reutov** [comment]
@desfire softaculous will be in 0.9.8-18

**Comment by desfire** [comment]
@anton-reutov and this will be just an update with fixes right? or is the 18 coming soon?

**Comment by anton-reutov** [comment]
Everyone who is testing, check the backups

**Comment by hwcltjn** [comment]
Will do - thanks Anton!
Anyone made a list of what they’re testing and feel like sharing?

**Comment by anton-reutov** [comment]
@gethprime 
@amaisonneuve 
What about the test? How the results ? Test servers will be working up to Dec 16

**Comment by amaisonneuve** [comment]
Ok here is my breakdown:

 

Add Domain
Delete Domain
Edit Domain
SSL Generate Cert
Add Email
Delete Email
Add FTP Account
Delete FTP Account
View ALL pages

 

My only comments are:

 

Why is the system opening new tabs for each action, this is something new and well not really wanted/needed.

SSL Cert with Lets Encrypt request fails

Everything else looks good.

**Comment by anton-reutov** [comment]
Write more about LE. Whats problem ?

**Comment by tim427** [comment]
@anton-reutov Your VPS is ending at 16th of december? What are the requirements? Maybe I can provide you guys a VPS :)

**Comment by anton-reutov** [comment]
@tim427 we're in a hurry. Therefore, the testing time is limited :))

**Comment by hwcltjn** [comment]
My test results:

✓ Add User
✓ Logged in as user (directly, not through admin)
✓ Add domain as user
✓ Add second domain as user
✓ Delete domain as user
✓ Create LE certificate
✓ Delete LE certificate
✓ Create mail domain
✓ Create several mail accounts
✓ Logged into mail accounts using RoundCube
✓ Delete mail accounts
✓ Created FTP accounts
✓ Access FTP accounts using FTP client
✓ Delete FTP accounts
✓ Created DNS domains
✓ Tested dns resolution with dig - local and remote
✓ Deleted DNS domains
✓ Create DB
✓ Delete DB
✓ Viewed all settings, graph, package, log, statistics, server, firewall, cron and update pages
✓ Backup creation

☓ Mailbox quota does not appear in RoundCube
☓ Vsftpd fail2ban jail is disabled by default
☓ MySQL port 3306, 5432 open by default

I did not test:
- Adding new IP
- Creating new package
- Adding new cronjob
- Restoring backup
- Shell access
- Adding/removing firewall rule
- Different hosting templates

**Comment by anton-reutov** [comment]
@hwcltjn do you check the backups on user lavel ?

**Comment by anton-reutov** [comment]
Who wrote about a separate page in Google chrome

**Comment by lukapaunovic** [comment]
> But that way old installer will be used, @anton-reutov
> If somebody already decided to use git version of code, then better to use installer from git (that is newer version).
> 
> If someone want to use installer from git, then:
> 
> ```
> cd $(mktemp -d)
> git clone git://github.com/serghey-rodin/vesta.git
> bash vesta/install/vst-install-rhel.sh #or other distro: vst-install-debian.sh or vst-install-ubuntu.sh
> yes | cp -rf vesta/* /usr/local/vesta
> ```

I just wanna correct this command.
You shouldn't have any output while running

`yes | cp -rf vesta/* /usr/local/vesta`

But if u do, it's because cp  is being already aliased to something like cp -i. Calling cp directly should work better:

`yes | /usr/bin/cp -rf vesta/* /usr/local/vesta`

And it produces no warnings.

**Comment by hwcltjn** [comment]
@anton-reutov yes, I created the backups at the user level.

**Comment by anton-reutov** [comment]
@hwcltjn ok, thank you for your help.

**Comment by desfire** [comment]
I have an unused VPS if someone needs it to test, I can give access.

1CPU
100GB
1 RAM

**Comment by mghadam** [comment]
any updates on this issue guys ?

**Comment by anton-reutov** [comment]
@mghadam about new release, or results of testing new templates ?

**Comment by mghadam** [comment]
@anton-reutov on releasing a new version of vestacp. @serghey-rodin vestacp needs an official team now to grow

**Comment by anton-reutov** [comment]
@mghadam, we start to a building packages. 0.9.8-18 will be in the early of next week. 
The team needs to doing a final testing. When the final testing is done we release a new version

**Comment by anton-reutov** [comment]
The final testing is beginning :)
Tomorrow i send to our dev team the link on test installer.

It's time to close this issue.

**Comment by alivelimeli** [comment]
Is final testing finished? Could you please update?

**Comment by anton-reutov** [comment]
@alivelimeli , Yes, almost
The Softaculous team has completed the test today.

**Comment by desfire** [comment]
I have installed the new version via git for tests, how do I install softaculous?

**Comment by anton-reutov** [comment]
Hi, guys, new version has been released today.
All questons on our forum.
With Happy New Year and thanks to all for the help.

**Comment by phez** [comment]
Congratulations on new version!

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by azzronulation** [comment]
Awesome! Is there anything particular I need to do before updating? Is there an easy way to backup configs etc etc?

**Comment by anton-reutov** [comment]
You can use this command for update VestaCP
`v-update-sys-vesta-all`

**Comment by darkworks** [comment]
nice but important thing missing ipv6 , support , i thought @madeITBelgium  work a lot on it , looks like its not merged yet
